### PR TITLE
This allow configurability of the service type instead of relying on …

### PR DESCRIPTION
**Pull request by mikebell90**
…a manifest

### CODE REVIEW

**Comment by mikebell90** [comment]
The . name chosen is compatible with Dmitry K's choice. We could allow a cascade.

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java [hunk 1]
```diff
@@ -38,7 +40,10 @@
     public void onApplicationEvent(ApplicationEvent event) {
         if (event instanceof ApplicationEnvironmentPreparedEvent) {
             ApplicationEnvironmentPreparedEvent castEvent = (ApplicationEnvironmentPreparedEvent) event;
-            String serviceName = castEvent.getSpringApplication().getMainApplicationClass().getPackage().getImplementationTitle();
+            String serviceName = castEvent.getEnvironment().getProperty(COMPONENT_NAME_KEY);
```
Comment (adamzr) [inline src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: Where does it get set?

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java [hunk 2]
```diff
@@ -28,6 +30,8 @@
 
 /* Registered via META-INF/spring.factories to capture early application lifecycle events */
 class ServerLoggingConfiguration implements ApplicationListener<ApplicationEvent> {
+    public static final String DMITRY_COMPONENT_NAME_KEY = "info.component"; // Taken from Dmitry's stack
+    public static final String OT_COMPONENT_NAME_KEY = "ot.component.id"; // Taken from Dmitry's stack
```
Comment (adamzr) [inline src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: Dimitry used 2 versions?

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java [hunk 3]
```diff
@@ -38,7 +42,12 @@
     public void onApplicationEvent(ApplicationEvent event) {
         if (event instanceof ApplicationEnvironmentPreparedEvent) {
             ApplicationEnvironmentPreparedEvent castEvent = (ApplicationEnvironmentPreparedEvent) event;
-            String serviceName = castEvent.getSpringApplication().getMainApplicationClass().getPackage().getImplementationTitle();
+            // Prefer OT, then Dmitry, then Manifest
```
Comment (adamzr) [inline src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: I thought that they're both from Dimitry?

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java (hunk 2 above)
Comment (adamzr) [inline src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: I'd prefer `component-id`

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java (hunk 3 above)
Comment (adamzr) [inline src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: Why not use the service name in the service info bean?

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java (hunk 2 above)
Comment (adamzr) [inline src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: Where do these get set?

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java [hunk 4]
```diff
@@ -38,7 +42,12 @@
...
+            String serviceName =
+                    Optional.ofNullable(castEvent.getEnvironment().getProperty(OT_COMPONENT_NAME_KEY))
+                            .orElse(Optional.ofNullable(castEvent.getEnvironment().getProperty(DMITRY_COMPONENT_NAME_KEY))
+                                    .orElse(castEvent.getSpringApplication().getMainApplicationClass().getPackage().getImplementationTitle())
```
Comment (adamzr) [inline src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: Add comment explaining how the manifest version gets set.

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java (hunk 2 above)
Comment (mikebell90) [reply src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: No, I was introducing a new one. Arguably more consistent with our naming schemae

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java (hunk 3 above)
Comment (mikebell90) [reply src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: Nope see above

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java (hunk 2 above)
Comment (mikebell90) [reply src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: I don't feel super strongly, but note if you don't have a root namespace, it's easy to get annoying collisions

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java (hunk 3 above)
Comment (mikebell90) [reply src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: You can't get it this early in lifecycle unfortunately. Remember the logger needs to be set up before the context is built, or we dont get bad wiring logging

File: src/main/java/com/opentable/server/ServerLoggingConfiguration.java (hunk 2 above)
Comment (mikebell90) [reply src/main/java/com/opentable/server/ServerLoggingConfiguration.java]: in any spring compliant config file etc.

**Pull request merged**